## Random Forest Wage Predictor

Here we take data of 4165 people in the US from 1976 to 1982 containing their wages, along with attributes such as marraige status, race, and sex. We train a random forest model to predict the wages of people in the testing set.

In [1]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn import tree
from sklearn.ensemble import RandomForestClassifier

#data is from https://vincentarelbundock.github.io/Rdatasets/doc/Ecdat/Wages.html

First we clean the data. Since the random forest uses logic trees, we only want data that can be represented in binary. 

In [2]:
df = pd.read_csv("Wages.csv")
median_wage = np.median(df['lwage'])

df['wage_high'] = df['lwage'].apply(lambda row: 1 if row > median_wage else 0)
df['wage_high'].value_counts()

df = df[['bluecol', 'ind', 'south', 'smsa', 'married', 'sex', 'union', 'black', 'wage_high']]
for i in df[['bluecol', 'ind', 'south', 'smsa', 'married', 'sex', 'union', 'black']]:
    df[i] = df[i].apply(lambda row: 1 if row == 'yes' else 1 if row == 'male' else 0)

y = df['wage_high'].to_numpy()

X = df
X.drop(['wage_high'], axis=1, inplace = True)

We split the data in to testing and training sets, train the model, and test it.

In [3]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state=2)

In [4]:
forest = RandomForestClassifier(random_state=2)
forest.fit(X_train, y_train)
forest.score(X_test, y_test)

0.7310924369747899

A score of 0.73 is not terrible for this sample data!